WarshipGame/Assets/Scripts/Gameplay/Camera/CameraController.cs

```diff
@@ -0,0 +1,175 @@
+using System.Collections;
+using UnityEngine;
+using Cinemachine;
+
+public class CameraController : MonoBehaviour
+{
+    [SerializeField] private CinemachineVirtualCamera CMVirtualCamera;
+    [SerializeField] private Camera Cam;
+
+    [Header("Panning")]
+    
+    [Tooltip("The speed in which the player can drag the camera across the game scene.")]
+    [SerializeField] private float PanningSpeed = 10f;
+    [Tooltip("The amount of Hexes in the X-axis on the map.")]
+    [SerializeField] private float MapSizeX = 8f;
+    [Tooltip("The amount of Hexes in the Z-axis on the map.")]
+    [SerializeField] private float MapSizeZ = 8f;
+    private Vector3 _cameraOrigin;
+    private Vector3 _dragOrigin;
+    private Vector3 _dragDirection;
+    
+    [Header("Zooming")]
+    
+    [Tooltip("The FOV of the camera will change to this value once a focus point is selected.")]
+    [SerializeField] private float FocusValue = 30f;
+    private GameObject _focusGameObject;
+    private float _focusBoundaries;
+    private bool _isFocussing;
+    
+    [Space]
+    [Tooltip("The Cameras FOV will always be in between these 2 values.")]
+    [SerializeField] private Vector2 MaxZoom = new(35f,60f);
+    [Tooltip("The Speed in which the player can zoom the camera FOV.")]
+    [SerializeField] private float ZoomSpeed = 9;
+    [Tooltip("The standard zoomFOV for this camera controller.")] 
+    [SerializeField] private float DefaultFOVValue = 45;
+    private float _currentFOVValue;
+    
+    private void Awake()
+    {
+        MapSizeX *= 0.866f;
+        MapSizeZ = MapSizeZ * 0.866f - (0.866f / 2);
+        
+        _currentFOVValue = DefaultFOVValue;
+        CMVirtualCamera.m_Lens.FieldOfView = _currentFOVValue;
+    }
+
+    private void FixedUpdate()
+    {
+        PanCamera();
+
+        if (Input.GetAxis("Mouse ScrollWheel") != 0 && !_isFocussing)
+        {
+            ZoomCamera();
```

Comment:
you check ```Input.GetAxis("Mouse ScrollWheel") != 0``` adn then when you run ```ZoomCamera()``` you call `````Input.GetAxis("Mouse ScrollWheel")``` into a local variable in the if statement make it a local variable and give it as a argument to ```ZoomCamera()``` that way we are using 1 input call less which is better for performance

Reply:
not really sure how to fix this cuz first is a check and second is the accual input